### ISSUE TITLE
UNION causes "ERROR:  invalid memory alloc request"

### ISSUE THREAD

**Issue by shawjef3** [GeoffMontee/tds_fdw#14]
Hello,
I'm using SQL Server 2008 R2 and PostgreSQL 9.4 beta 3. UNION causes an error. UNION ALL works.

```SQL
CREATE FOREIGN TABLE public.one (
  one int NOT NULL
)
SERVER jeff OPTIONS (database 'master', query 'SELECT 1')
```
```SQL
SELECT * FROM one
UNION
SELECT * FROM one
```

```
NOTICE:  DB-Library notice: Msg #: 5701, Msg state: 2, Msg: Changed database context to 'master'., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5703, Msg state: 1, Msg: Changed language setting to us_english., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5701, Msg state: 1, Msg: Changed database context to 'master'., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5701, Msg state: 2, Msg: Changed database context to 'master'., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5703, Msg state: 1, Msg: Changed language setting to us_english., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5701, Msg state: 1, Msg: Changed database context to 'master'., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5701, Msg state: 2, Msg: Changed database context to 'master'., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5703, Msg state: 1, Msg: Changed language setting to us_english., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5701, Msg state: 1, Msg: Changed database context to 'master'., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5701, Msg state: 2, Msg: Changed database context to 'master'., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5703, Msg state: 1, Msg: Changed language setting to us_english., Server: jeff, Process: , Line: 1, Level: 0
NOTICE:  DB-Library notice: Msg #: 5701, Msg state: 1, Msg: Changed database context to 'master'., Server: jeff, Process: , Line: 1, Level: 0


ERROR:  invalid memory alloc request size 18446744073709551568
********** Error **********

ERROR: invalid memory alloc request size 18446744073709551568
SQL state: XX000
```

```SQL
SELECT * FROM one
UNION ALL
SELECT * FROM one
```

```
1
1
```

**Comment by GeoffMontee** [comment]
This actually seemed to be caused by `dbcount()` returning meaningless values for some reason, which makes it a duplicate of issue #15. This is fixed in the latest commit.

Commit: https://github.com/GeoffMontee/tds_fdw/commit/fb5b96d32b6c7dabfafee8c78ca6437b57d7f1ef

**Comment by GeoffMontee** [comment]
Fixed in version 1.0.2.
